syntax = "proto3";
package gorm;
option go_package = "gorm/gorm";
import "gorm-db.proto";


enum GORM_PB_TABLE_INDEX
{
    GORM_PB_TABLE_IDX_MIN__ = 0;
	GORM_PB_TABLE_IDX_CURRENCY = 1;
	GORM_PB_TABLE_IDX_CYCLEEVENT = 2;
	GORM_PB_TABLE_IDX_EQUIP = 3;
	GORM_PB_TABLE_IDX_FOOD = 4;
	GORM_PB_TABLE_IDX_HERO = 5;
	GORM_PB_TABLE_IDX_LOGIN_LOG = 6;
	GORM_PB_TABLE_IDX_MAIL = 7;
	GORM_PB_TABLE_IDX_MATERIAL = 8;
	GORM_PB_TABLE_IDX_NPC = 9;
	GORM_PB_TABLE_IDX_RES_LOG = 10;
	GORM_PB_TABLE_IDX_INTERACTION = 11;
	GORM_PB_TABLE_IDX_ROLES = 12;
	GORM_PB_TABLE_IDX_ROLE = 13;
	GORM_PB_TABLE_IDX_SCENE = 14;
	GORM_PB_TABLE_IDX_SKILL = 15;
	GORM_PB_TABLE_IDX_PTUSER = 16;
	GORM_PB_TABLE_IDX_USER = 17;
	GORM_PB_TABLE_IDX_MAINLINETASK = 18;
	GORM_PB_TABLE_IDX_PUBMAIL = 19;
	GORM_PB_TABLE_IDX_MAX__ = 20;
}

enum GORM_PB_CURRENCY_FIELD_INDEX
{
	GORM_PB_FIELD_CURRENCY_VERSION = 0;
	GORM_PB_FIELD_CURRENCY_ROLEID = 1;
	GORM_PB_FIELD_CURRENCY_SNAKEID = 2;
	GORM_PB_FIELD_CURRENCY_CURRENCY1 = 3;
	GORM_PB_FIELD_CURRENCY_CURRENCY2 = 4;
	GORM_PB_FIELD_CURRENCY_CURRENCY3 = 5;
	GORM_PB_FIELD_CURRENCY_CURRENCY4 = 6;
}

enum GORM_PB_CYCLEEVENT_FIELD_INDEX
{
	GORM_PB_FIELD_CYCLEEVENT_VERSION = 0;
	GORM_PB_FIELD_CYCLEEVENT_ROLEID = 1;
	GORM_PB_FIELD_CYCLEEVENT_EVENTS = 2;
}

enum GORM_PB_EQUIP_FIELD_INDEX
{
	GORM_PB_FIELD_EQUIP_VERSION = 0;
	GORM_PB_FIELD_EQUIP_ROLEID = 1;
	GORM_PB_FIELD_EQUIP_SNAKEID = 2;
	GORM_PB_FIELD_EQUIP_EQUIP1 = 3;
	GORM_PB_FIELD_EQUIP_EQUIP2 = 4;
	GORM_PB_FIELD_EQUIP_EQUIP3 = 5;
	GORM_PB_FIELD_EQUIP_EQUIP4 = 6;
}

enum GORM_PB_FOOD_FIELD_INDEX
{
	GORM_PB_FIELD_FOOD_VERSION = 0;
	GORM_PB_FIELD_FOOD_ROLEID = 1;
	GORM_PB_FIELD_FOOD_SNAKEID = 2;
	GORM_PB_FIELD_FOOD_FOOD1 = 3;
	GORM_PB_FIELD_FOOD_FOOD2 = 4;
	GORM_PB_FIELD_FOOD_FOOD3 = 5;
	GORM_PB_FIELD_FOOD_FOOD4 = 6;
}

enum GORM_PB_HERO_FIELD_INDEX
{
	GORM_PB_FIELD_HERO_VERSION = 0;
	GORM_PB_FIELD_HERO_ROLEID = 1;
	GORM_PB_FIELD_HERO_SNAKEID = 2;
	GORM_PB_FIELD_HERO_HEROS = 3;
	GORM_PB_FIELD_HERO_TEAMS = 4;
}

enum GORM_PB_LOGIN_LOG_FIELD_INDEX
{
	GORM_PB_FIELD_LOGIN_LOG_VERSION = 0;
	GORM_PB_FIELD_LOGIN_LOG_ROLEID = 1;
	GORM_PB_FIELD_LOGIN_LOG_ACTION = 2;
	GORM_PB_FIELD_LOGIN_LOG_ONLINE = 3;
	GORM_PB_FIELD_LOGIN_LOG_IP = 4;
	GORM_PB_FIELD_LOGIN_LOG_APPID = 5;
	GORM_PB_FIELD_LOGIN_LOG_CREATETIME = 6;
}

enum GORM_PB_MAIL_FIELD_INDEX
{
	GORM_PB_FIELD_MAIL_VERSION = 0;
	GORM_PB_FIELD_MAIL_ROLEID = 1;
	GORM_PB_FIELD_MAIL_MAIL1 = 2;
	GORM_PB_FIELD_MAIL_MAIL2 = 3;
	GORM_PB_FIELD_MAIL_MAIL3 = 4;
	GORM_PB_FIELD_MAIL_MAIL4 = 5;
	GORM_PB_FIELD_MAIL_MAIL5 = 6;
	GORM_PB_FIELD_MAIL_LASTMAILID = 7;
	GORM_PB_FIELD_MAIL_SNAKEID = 8;
}

enum GORM_PB_MATERIAL_FIELD_INDEX
{
	GORM_PB_FIELD_MATERIAL_VERSION = 0;
	GORM_PB_FIELD_MATERIAL_ROLEID = 1;
	GORM_PB_FIELD_MATERIAL_SNAKEID = 2;
	GORM_PB_FIELD_MATERIAL_MATERIAL1 = 3;
	GORM_PB_FIELD_MATERIAL_MATERIAL2 = 4;
	GORM_PB_FIELD_MATERIAL_MATERIAL3 = 5;
	GORM_PB_FIELD_MATERIAL_MATERIAL4 = 6;
}

enum GORM_PB_NPC_FIELD_INDEX
{
	GORM_PB_FIELD_NPC_VERSION = 0;
	GORM_PB_FIELD_NPC_ROLEID = 1;
	GORM_PB_FIELD_NPC_SNAKEID = 2;
	GORM_PB_FIELD_NPC_NPCS = 3;
}

enum GORM_PB_RES_LOG_FIELD_INDEX
{
	GORM_PB_FIELD_RES_LOG_VERSION = 0;
	GORM_PB_FIELD_RES_LOG_ROLEID = 1;
	GORM_PB_FIELD_RES_LOG_ACTION = 2;
	GORM_PB_FIELD_RES_LOG_CFGTYPE = 3;
	GORM_PB_FIELD_RES_LOG_CFGID = 4;
	GORM_PB_FIELD_RES_LOG_DELTA = 5;
	GORM_PB_FIELD_RES_LOG_VALUE = 6;
	GORM_PB_FIELD_RES_LOG_CREATETIME = 7;
}

enum GORM_PB_INTERACTION_FIELD_INDEX
{
	GORM_PB_FIELD_INTERACTION_VERSION = 0;
	GORM_PB_FIELD_INTERACTION_ROLEID = 1;
	GORM_PB_FIELD_INTERACTION_SNAKEID = 2;
	GORM_PB_FIELD_INTERACTION_INTERACTION1 = 3;
	GORM_PB_FIELD_INTERACTION_INTERACTION2 = 4;
	GORM_PB_FIELD_INTERACTION_INTERACTION3 = 5;
	GORM_PB_FIELD_INTERACTION_INTERACTION4 = 6;
}

enum GORM_PB_ROLES_FIELD_INDEX
{
	GORM_PB_FIELD_ROLES_VERSION = 0;
	GORM_PB_FIELD_ROLES_ROLEID = 1;
	GORM_PB_FIELD_ROLES_USERID = 2;
	GORM_PB_FIELD_ROLES_WORLDID = 3;
	GORM_PB_FIELD_ROLES_DBID = 4;
	GORM_PB_FIELD_ROLES_NAME = 5;
	GORM_PB_FIELD_ROLES_CHARID = 6;
	GORM_PB_FIELD_ROLES_FACE = 7;
	GORM_PB_FIELD_ROLES_CREATETIME = 8;
}

enum GORM_PB_ROLE_FIELD_INDEX
{
	GORM_PB_FIELD_ROLE_VERSION = 0;
	GORM_PB_FIELD_ROLE_ROLEID = 1;
	GORM_PB_FIELD_ROLE_ROLENAME = 2;
	GORM_PB_FIELD_ROLE_LEVEL = 3;
	GORM_PB_FIELD_ROLE_EXP = 4;
	GORM_PB_FIELD_ROLE_CHARACTERID = 5;
	GORM_PB_FIELD_ROLE_GOLD = 6;
	GORM_PB_FIELD_ROLE_OFFLINETIME = 7;
	GORM_PB_FIELD_ROLE_INITED = 8;
	GORM_PB_FIELD_ROLE_CREATETIME = 9;
	GORM_PB_FIELD_ROLE_FACE = 10;
	GORM_PB_FIELD_ROLE_ONLINE = 11;
	GORM_PB_FIELD_ROLE_WORLDID = 12;
	GORM_PB_FIELD_ROLE_PTTYPE = 13;
	GORM_PB_FIELD_ROLE_USERID = 14;
	GORM_PB_FIELD_ROLE_PROCEEDSLV = 15;
}

enum GORM_PB_SCENE_FIELD_INDEX
{
	GORM_PB_FIELD_SCENE_VERSION = 0;
	GORM_PB_FIELD_SCENE_ROLEID = 1;
	GORM_PB_FIELD_SCENE_SCENEID = 2;
	GORM_PB_FIELD_SCENE_COLLECTS = 3;
	GORM_PB_FIELD_SCENE_SEASON = 4;
}

enum GORM_PB_SKILL_FIELD_INDEX
{
	GORM_PB_FIELD_SKILL_VERSION = 0;
	GORM_PB_FIELD_SKILL_ROLEID = 1;
	GORM_PB_FIELD_SKILL_SNAKEID = 2;
	GORM_PB_FIELD_SKILL_SKILL1 = 3;
	GORM_PB_FIELD_SKILL_SKILL2 = 4;
	GORM_PB_FIELD_SKILL_SKILL3 = 5;
	GORM_PB_FIELD_SKILL_SKILL4 = 6;
}

enum GORM_PB_PTUSER_FIELD_INDEX
{
	GORM_PB_FIELD_PTUSER_VERSION = 0;
	GORM_PB_FIELD_PTUSER_PTID = 1;
	GORM_PB_FIELD_PTUSER_PTTYPE = 2;
	GORM_PB_FIELD_PTUSER_USERID = 3;
}

enum GORM_PB_USER_FIELD_INDEX
{
	GORM_PB_FIELD_USER_VERSION = 0;
	GORM_PB_FIELD_USER_USERID = 1;
	GORM_PB_FIELD_USER_PTID = 2;
	GORM_PB_FIELD_USER_PTTYPE = 3;
	GORM_PB_FIELD_USER_CREATETIME = 4;
	GORM_PB_FIELD_USER_ROLESDATA = 5;
}

enum GORM_PB_MAINLINETASK_FIELD_INDEX
{
	GORM_PB_FIELD_MAINLINETASK_VERSION = 0;
	GORM_PB_FIELD_MAINLINETASK_ROLEID = 1;
	GORM_PB_FIELD_MAINLINETASK_RUNNINGTASK = 2;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK0 = 3;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK1 = 4;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK2 = 5;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK3 = 6;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK4 = 7;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK5 = 8;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK6 = 9;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK7 = 10;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK8 = 11;
	GORM_PB_FIELD_MAINLINETASK_COMPLEATETASK9 = 12;
	GORM_PB_FIELD_MAINLINETASK_UNLOCKFUNCS = 13;
}

enum GORM_PB_PUBMAIL_FIELD_INDEX
{
	GORM_PB_FIELD_PUBMAIL_VERSION = 0;
	GORM_PB_FIELD_PUBMAIL_MAILID = 1;
	GORM_PB_FIELD_PUBMAIL_ADDRESSER = 2;
	GORM_PB_FIELD_PUBMAIL_TITLE = 3;
	GORM_PB_FIELD_PUBMAIL_BODY = 4;
	GORM_PB_FIELD_PUBMAIL_ROLES = 5;
	GORM_PB_FIELD_PUBMAIL_ATTACHMENT = 6;
	GORM_PB_FIELD_PUBMAIL_DT = 7;
	GORM_PB_FIELD_PUBMAIL_DEL = 8;
}


enum GORM_PB_COLUMN_TYPE
{
	GORM_PB_COLUMN_TYPE_INVALID = 0;
	GORM_PB_COLUMN_TYPE_INT		= 1;
	GORM_PB_COLUMN_TYPE_UINT 	= 2;
	GORM_PB_COLUMN_TYPE_DOUBLE	= 3;
	GORM_PB_COLUMN_TYPE_STRING	= 4;
	GORM_PB_COLUMN_TYPE_INT8	= 5;
	GORM_PB_COLUMN_TYPE_INT16	= 6;
	GORM_PB_COLUMN_TYPE_INT32	= 7;
	GORM_PB_COLUMN_TYPE_INT64	= 8;
	GORM_PB_COLUMN_TYPE_UINT8	= 9;
	GORM_PB_COLUMN_TYPE_UINT16	= 10;
	GORM_PB_COLUMN_TYPE_UINT32	= 11;
	GORM_PB_COLUMN_TYPE_UINT64	= 12;
	GORM_PB_COLUMN_TYPE_BLOB	= 13;
	GORM_PB_COLUMN_TYPE_CHAR	= 14;
	GORM_PB_COLUMN_TYPE_BYTES   = 15;
}

message GORM_PB_COLUMN_VALUE
{
	GORM_PB_COLUMN_TYPE	type 		= 1;
	fixed64				uintvalue 	= 2;
	sfixed64				intvalue 	= 3;
	double					doublevalue = 4;
	string					stringvalue	= 5;
}

message GORM_PB_COLUMN
{
	string  				name 	= 1;				// 列名
	GORM_PB_COLUMN_VALUE 	value 	= 2;	// 存储的数据
}


message GORM_PB_CUSTEM_COLUMNS
{
	repeated	GORM_PB_COLUMN	columns 	= 1;
}
message GORM_PB_TABLE
{
	sfixed32 TableId = 1;
	GORM_PB_CUSTEM_COLUMNS	custom_columns = 2;
	 GORM_PB_Table_currency currency = 3;
	 GORM_PB_Table_cycleevent cycleevent = 4;
	 GORM_PB_Table_equip equip = 5;
	 GORM_PB_Table_food food = 6;
	 GORM_PB_Table_hero hero = 7;
	 GORM_PB_Table_login_log login_log = 8;
	 GORM_PB_Table_mail mail = 9;
	 GORM_PB_Table_material material = 10;
	 GORM_PB_Table_npc npc = 11;
	 GORM_PB_Table_res_log res_log = 12;
	 GORM_PB_Table_interaction interaction = 13;
	 GORM_PB_Table_roles roles = 14;
	 GORM_PB_Table_role role = 15;
	 GORM_PB_Table_scene scene = 16;
	 GORM_PB_Table_skill skill = 17;
	 GORM_PB_Table_ptuser ptuser = 18;
	 GORM_PB_Table_user user = 19;
	 GORM_PB_Table_mainlinetask mainlinetask = 20;
	 GORM_PB_Table_pubmail pubmail = 21;
}

message GORM_PB_TABLES
{
	repeated GORM_PB_Table_currency currency = 1;
	repeated GORM_PB_Table_cycleevent cycleevent = 2;
	repeated GORM_PB_Table_equip equip = 3;
	repeated GORM_PB_Table_food food = 4;
	repeated GORM_PB_Table_hero hero = 5;
	repeated GORM_PB_Table_login_log login_log = 6;
	repeated GORM_PB_Table_mail mail = 7;
	repeated GORM_PB_Table_material material = 8;
	repeated GORM_PB_Table_npc npc = 9;
	repeated GORM_PB_Table_res_log res_log = 10;
	repeated GORM_PB_Table_interaction interaction = 11;
	repeated GORM_PB_Table_roles roles = 12;
	repeated GORM_PB_Table_role role = 13;
	repeated GORM_PB_Table_scene scene = 14;
	repeated GORM_PB_Table_skill skill = 15;
	repeated GORM_PB_Table_ptuser ptuser = 16;
	repeated GORM_PB_Table_user user = 17;
	repeated GORM_PB_Table_mainlinetask mainlinetask = 18;
	repeated GORM_PB_Table_pubmail pubmail = 19;
}

